packages/grid/_modules_/grid/models/gridRows.ts

```diff
@@ -1,10 +1,11 @@
 export type GridRowsProp = Readonly<GridRowData[]>;
-export type GridRowData = { [key: string]: any };
+export type GridRowType = { [key: string]: any };
+export type GridRowData<RowType = GridRowType> = RowType;
 
 /**
  * The key value object representing the data of a row.
  */
-export type GridRowModel = GridRowData;
+export type GridRowModel<RowType = any> = GridRowData<RowType>;
```

Comment:
I think we should avoid doing ` export type GridRowData<RowType = GridRowType> = RowType;` 
Because we would have 3 types basically doing the same thing `GridRowData`, `GridRowType` and `GridRowModel`

I would do the following:

```tsx
export type GridRowDefaultData = { [key: string]: any };

export type GridRowModel<RowType extends GridRowDefaultData = GridRowDefaultData> = GridRowDefaultData
```

Reply:
Yeah thanks, I was also thinking about what to name it. `GridRowDefaultData` sounds good.

Reply:
The reason for having `GridRowModel` and `GridRowData` is that, in the [past](https://github.com/mui-org/material-ui-x/pull/1377/files#diff-7f232d14dba7e4e30d0acb33b58b964bf8a8f9dd2c38b47981d98d976a336b4aL7), the first one would always contain the row id. The latter one could be removed.